src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -294,23 +229,9 @@ def update_stats(self, area, simulation_status, progress_info, sim_state):
         self.update_results_area_uuids(area)
         self.update_offer_bid_trade()
 
-    def update_area_aggregated_stats(self, area_dict):
-        self._merge_cumulative_bills_into_bills_for_market_info(area_dict)
-        for child in area_dict['children']:
-            self.update_area_aggregated_stats(child)
-
     def update_offer_bid_trade(self):
         if self.current_market_time_slot_str == "":
             return
         for area_uuid, area_result in self.flattened_area_core_stats_dict.items():
             self.bids_offers_trades[area_uuid] = \
                 {k: area_result[k] for k in ('offers', 'bids', 'trades')}
-
-    def _merge_cumulative_bills_into_bills_for_market_info(self, area_dict):
-        bills = self.market_bills.bills_redis_results[area_dict['uuid']]
-        bills.update({
-            "penalty_cost":
-                self.cumulative_bills.cumulative_bills_results[area_dict['uuid']]["penalties"],
-            "penalty_energy":
-                self.cumulative_bills.cumulative_bills_results[area_dict['uuid']]["penalty_energy"]
-        })
```

Comment:
So, you removed this already, because you know that this is not needed any more? I dont see the penalties added to the bills somewhere else. Can you point me to it please?

Reply:
It is not called from anywhere in the code, that's why I removed it, and I (apparently) falsely assumed that the penalties feature is deprecated from the market bills. Also, after some brief investigation, there are some integration tests steps (in kpi.py, for instance `"device '{device_name}' reports penalties of {penalty_energy} kWh`) that involve penalties but are never used in any test. Since apparently this is not the case, let me check why or when were all these removed and how to reinstate them.

Reply:
As a summary of an offline discussion, penalties in market bills have been deprecated (they only exist in cumulative bills now) , already dealt with in this PR https://github.com/gridsingularity/d3a/pull/740. Meanwhile there are some other parts of the code that can be removed (specifically integration tests steps), will add these to the PR.